lib/utils/get-pin-details.ts

```diff
@@ -0,0 +1,113 @@
+import type { Parser as ParserType, Pin } from ".././dsn-pcb/types"
+import Debug from "debug"
+import { getPinNum } from "lib/utils/get-pin-number"
+
+const debug = Debug("dsn-converter:get-pin-details")
+export function getPinDetails(nodes: any[]): Pin | null {
```

Comment:
can you improve the type here? it was originally like AstNode? I feel like one way we can make things more maintanble is improving types or at least preventing type regressions